code/modules/surgery/bodyparts/species_parts/genetics_bodyparts.dm

```diff
@@ -0,0 +1,222 @@
+		new /obj/effect/decal/cleanable/glass(get_step(bunny, bunny.dir))
+	for(var/obj/machinery/door/window/windoor in get_turf(bunny))
+		windoor.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,15), damagetype = BRUTE, wound_bonus = 15, bare_wound_bonus = 25, sharpness = SHARP_EDGED, attack_direction = get_dir(windoor, oldloc))
+		new /obj/effect/decal/cleanable/glass(get_step(bunny, bunny.dir))
+	for(var/obj/structure/grille/grille in get_turf(bunny))
+		grille.shock(bunny, 70)
+		grille.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,10), damagetype = BRUTE, wound_bonus = 5, bare_wound_bonus = 15, attack_direction = get_dir(grille, oldloc))
+		new /obj/effect/decal/cleanable/generic(get_step(bunny, bunny.dir))
+
+/datum/action/cooldown/spell/bunny_hop/proc/break_grilles(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/bunny = mover
+	for(var/obj/structure/grille/grille in get_turf(bunny))
+		grille.shock(bunny, 70)
+		grille.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,10), damagetype = BRUTE, wound_bonus = 5, bare_wound_bonus = 15, attack_direction = get_dir(grille, oldloc))
+		new /obj/effect/decal/cleanable/generic(get_step(bunny, bunny.dir))
+
+///Ends the jump
+/datum/action/cooldown/spell/bunny_hop/proc/end_jump(mob/living/jumper)
+	jumper.remove_filter(BUNNY_HOP)
+	jumper.layer = initial(jumper.layer)
+	jumper.pass_flags = initial(jumper.pass_flags)
+	REMOVE_TRAIT(jumper, TRAIT_SILENT_FOOTSTEPS, BUNNY_HOP)
+	REMOVE_TRAIT(jumper, TRAIT_MOVE_FLYING, BUNNY_HOP)
+	new /obj/effect/temp_visual/mook_dust(get_turf(jumper))
+	UnregisterSignal(jumper, COMSIG_MOVABLE_MOVED)
+
+/obj/item/bodypart/leg/left/digitigrade/bunny
+	name = "rabbit left leg"
+	desc = "Helps you jump!"
+	icon_greyscale = 'icons/mob/human/species/misc/genetics_limbs.dmi'
+	base_limb_id = BODYPART_ID_RABBIT
+	var/datum/action/cooldown/spell/bunny_hop/jumping_power
+
+/obj/item/bodypart/leg/left/digitigrade/bunny/try_attach_limb(mob/living/carbon/new_head_owner, special)
+	. = ..()
+	if(!.)
+		return
+	var/potential_action = locate(/datum/action/cooldown/spell/bunny_hop) in new_head_owner.actions
+	if(potential_action)
+		jumping_power = potential_action
+	else
+		jumping_power = new(src)
+		jumping_power.background_icon_state = "bg_tech_blue"
+		jumping_power.base_background_icon_state = jumping_power.background_icon_state
+		jumping_power.active_background_icon_state = "[jumping_power.base_background_icon_state]_active"
+		jumping_power.overlay_icon_state = "bg_tech_blue_border"
+		jumping_power.active_overlay_icon_state = null
+		jumping_power.panel = "Genetic"
+		jumping_power.Grant(new_head_owner)
+	new_head_owner.AddElement(/datum/element/waddling, WADDLE_HOP)
```

Comment:
Assume you'd have run into it in testing but I am surprised this doesn't runtime, I'd have expected that the second leg would trigger a signal overridden runtime as it adds the element again.